packages/task/src/common/task-protocol.ts

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (C) 2017 Ericsson and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import { JsonRpcServer } from '@theia/core/lib/common/messaging/proxy-factory';
+import { RawProcessOptions } from '@theia/process/lib/node/raw-process';
+import { TerminalProcessOptions } from '@theia/process/lib/node/terminal-process';
+
+export const taskPath = '/services/task';
+
+export const TaskServer = Symbol('TaskServer');
+export const TaskClient = Symbol('TaskClient');
+
+export type ProcessType = 'terminal' | 'raw';
+
+export interface TaskInfo {
+    /** internal unique task id */
+    taskId: number,
+    /** terminal id. Defined if task is run as a terminal process */
+    terminalId?: number,
+    /** internal unique process id */
+    processId?: number,
+    /** OS PID of the process running the task */
+    osProcessId: number,
+    /** The command used to start this task */
+    command: string,
+    /** task label */
+    label: string,
+    /** context that was passed as part of task creation, if any */
+    ctx?: string
+}
+
+export interface TaskOptions {
+    /** A label that uniquely identifies a task configuration */
+    label: string,
+    /** 'raw' or 'terminal' */
+    processType: ProcessType,
```

Comment:
should be optional (with terminal as default?)

Reply:
done